pkg/dyninst/integration_test.go

```diff
@@ -194,51 +200,126 @@ func testDyninst(
+	require.NoError(t, err)
+
+	goDebugSections, err := moduledata.GoDebugSections(mef)
+	require.NoError(t, err)
+	defer func() { require.NoError(t, goDebugSections.Close()) }()
+
+	symbolTable, err := gosym.ParseGoSymbolTable(
+		goDebugSections.PcLnTab.Data,
+		goDebugSections.GoFunc.Data,
+		moduledata.Text,
+		moduledata.EText,
+		moduledata.MinPC,
+		moduledata.MaxPC,
+		goVersion,
+	)
+	require.NoError(t, err)
+	symbolicator := symbol.NewGoSymbolicator(symbolTable)
+	require.NotNil(t, symbolicator)
+
+	cachingSymbolicator, err := symbol.NewCachingSymbolicator(symbolicator, 10000)
+	require.NotNil(t, symbolicator)
+	require.NoError(t, err)
+
+	decoder, err := decode.NewDecoder(sink.irp, cachingSymbolicator)
 	require.NoError(t, err)
 	b := []byte{}
 	decodeOut := bytes.NewBuffer(b)
 	for _, msg := range read {
-		event := msg.Event()
+		event := decode.Event{
+			Event:       msg.Event(),
+			ServiceName: service,
+		}
 		err = decoder.Decode(event, decodeOut)
 		require.NoError(t, err)
-		header, err := event.Header()
-		require.NoError(t, err)
-
-		// Purge stack fraames
-		tmpMap := map[string]any{}
-		err = json.Unmarshal(decodeOut.Bytes(), &tmpMap)
-		require.NoError(t, err)
-		require.Equal(t, uint32(len(event)), header.Data_byte_len)
-		t.Logf("message header: %#v", *header)
-		if header.Stack_byte_len > 0 {
-			stackPCs, err := event.StackPCs()
-			require.NoError(t, err)
-			t.Logf("stack: %x", stackPCs)
-		}
-
-		if _, ok := tmpMap["stack_frames"]; !ok {
-			t.Error("No stack frames in output")
-		} else {
-			tmpMap["stack_frames"] = ""
-		}
 
-		clearAddressFields(tmpMap)
+		fmt.Println(decodeOut.String())
 
-		purged, err := json.Marshal(tmpMap)
-		assert.NoError(t, err)
+		t.Logf("Decoded output: \n%s\n", decodeOut.String())
+		redacted := redactJSON(t, decodeOut.Bytes())
+		t.Logf("Redacted output: \n%s\n", string(redacted))
 
 		outputToCompare := expOut[probes[0].GetID()]
-		assert.JSONEq(t, outputToCompare, string(purged))
+		assert.JSONEq(t, outputToCompare, string(redacted))
 
 		if saveOutput, _ := strconv.ParseBool(os.Getenv("REWRITE")); saveOutput {
-			expOut[probes[0].GetID()] = string(purged)
+			expOut[probes[0].GetID()] = string(redacted)
 			saveActualOutputOfProbes(t, service, expOut)
 		}
 	}
 }
 
+type jsonRedactor struct {
+	matches     func(ptr jsontext.Pointer) bool
+	replacement jsontext.Value
+}
+
+func redactPtr(toMatch string, replacement jsontext.Value) jsonRedactor {
+	return jsonRedactor{
+		matches: func(ptr jsontext.Pointer) bool {
+			return string(ptr) == toMatch
+		},
+		replacement: replacement,
+	}
+}
+
+func redactPtrPrefixSuffix(prefix, suffix string, replacement jsontext.Value) jsonRedactor {
+	return jsonRedactor{
+		matches: func(ptr jsontext.Pointer) bool {
+			return strings.HasPrefix(string(ptr), prefix) && strings.HasSuffix(string(ptr), suffix)
+		},
+		replacement: replacement,
+	}
+}
+
+var redactors = []jsonRedactor{
+	redactPtr("/debugger/snapshot/stack", []byte(`"<stack>"`)),
```

Comment:
maybe for the stacks we just redact some of the parts of the frame data? It seems to me like the function names shouldn't change and it's nice to see them.